plugins/data-management/src/HierarchicalTrackSelectorWidget/components/Contents.js

```diff
@@ -36,54 +35,19 @@ function Contents({
     hierarchy = hierarchy.get(pathEntry) || new Map()
   })
 
-  const initialTrackConfigurations = []
-  const initialCategories = []
+  const trackConfigurations = []
+  const categories = []
   Array.from(hierarchy)
-    .slice(0, 50)
+    .slice(0)
     .forEach(([name, contents]) => {
       if (contents.trackId) {
-        initialTrackConfigurations.push(contents)
+        trackConfigurations.push(contents)
       } else {
-        initialCategories.push([name, contents])
+        categories.push([name, contents])
       }
     })
-  const [categories, setCategories] = useState(initialCategories)
-  const [trackConfigurations, setTrackConfigurations] = useState(
-    initialTrackConfigurations,
-  )
-
-  useEffect(() => {
-    function loadMoreTracks() {
-      const numLoaded = categories.length + trackConfigurations.length
-      const loadedTrackConfigurations = []
-      const loadedCategories = []
-      Array.from(hierarchy)
-        .slice(numLoaded, numLoaded + 10)
-        .forEach(([name, contents]) => {
-          if (contents.trackId) {
-            loadedTrackConfigurations.push(contents)
-          } else {
-            loadedCategories.push([name, contents])
-          }
-        })
-      setCategories(categories.concat(loadedCategories))
-      setTrackConfigurations(
-        trackConfigurations.concat(loadedTrackConfigurations),
-      )
-    }
-
-    const handle = requestIdleCallback(loadMoreTracks)
-
-    return function cleanup() {
-      cancelIdleCallback(handle)
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [hierarchy.size, categories.length, trackConfigurations.length])
-
```

Comment:
Any reason the partial loading of tracks was removed? This was an attempt to make long track selector lists more responsive.

Reply:
Caused failure when a track was deleted, appeared to remain in the useState and get rendered after track deletion and caused a reference error. I thought track deletion would be a good demo to get into place and this was the only way I got it to work

Reply:
I had seen this at least once before where the interaction of usestate and observer components resulted in some stale values but can't remember when. Could try to reproduce